File: python/Crud.py
from Human import Human

class Crud:
    def __init__(self):
        self.__listHumans = []
    
    def tambah(self):
        print()
        id = input("Masukkan id      : ")
        nama = input("Masukkan nama    : ")
        bidang = input("Masukkan bidang  : ")
        partai = input("Masukkan partai  : ")
        temp = Human(id,nama,bidang,partai)
        self.__listHumans.append(temp)
        print( nama + " berhasil ditambahkan!\n")
    
    def ubah(self):
        print()
        if len(self.__listHumans)==0:
            print("Data tidak ada")
        else:
            print("Masukkan indeks nama")
            for i in range(len(self.__listHumans)):
                print(str(i+1)+". "+self.__listHumans[i].getNama())
		
            print()
            masukkan = int(input())
            print()
            
            if masukkan<=len(self.__listHumans) and masukkan > 0:
                print("Masukkan indeks isi")
                print("1. nama")
                print("2. id")
                print("3. bidang")
                print("4. partai")
                print()
				
                ubahB = int(input())
                print()
                
				
                if ubahB==1:
                    print("Masukkan nama : ")
                    nama = input()
                    self.__listHumans[masukkan-1].setNama(nama)
                elif ubahB==2:
                    print("Masukkan id : ")
                    id = input()
                    self.__listHumans[masukkan-1].setId(id)
                elif ubahB==3:
                    print("Masukkan bidang : ")
                    bidang = input()
                    self.__listHumans[masukkan-1].setBidang(bidang)
                elif ubahB==4:
                    print("Masukkan partai : ")
                    partai = input()
                    self.__listHumans[masukkan-1].setPartai(partai)
                else:
                    print("Indeks tidak valid")
            else:
                print("Indeks tidak valid")
			
		
        print()
	
    def hapus(self):
        print()
        if len(self.__listHumans)==0 :
            print("Data tidak ada")
        else:
            print("Masukkan indeks nama")
            for i in range(len(self.__listHumans)):
                print(str(i+1)+". "+self.__listHumans[i].getNama())
            
            print()
            masukkan = int(input())
            print()
            
            if masukkan<=len(self.__listHumans) and masukkan >0:
                print(self.__listHumans[masukkan-1].getNama() + " berhasil dihapus!")
                self.__listHumans.pop(masukkan-1)
            else:
                print("Indeks tidak valid")
            
        
        print()
	
    def tampil(self):
        print()
        if len(self.__listHumans)==0 :
            print("Data tidak ada", end="")
        else:
            maxNama=4
            maxid=2
            maxbidang=6
            maxpartai=6
            for i in range(len(self.__listHumans)):
                if(len(self.__listHumans[i].getNama())>maxNama):
                    maxNama = len(self.__listHumans[i].getNama())
                
                if(len(self.__listHumans[i].getId())>maxid):
                    maxid = len(self.__listHumans[i].getId())
                
                if(len(self.__listHumans[i].getBidang())>maxbidang):
                    maxbidang = len(self.__listHumans[i].getBidang())
                
                if(len(self.__listHumans[i].getPartai())>maxpartai):
                    maxpartai = len(self.__listHumans[i].getPartai())
                
            
            for j in range(maxNama+maxid+maxbidang+maxpartai+13):
                print("-", end="")
            print()
            
            print("| id", end="")
            for j in range(2,maxid):
                print(" ", end="")
            print(" | Nama", end="")
            for j in range(4,maxNama):
                print(" ", end="")
            print(" | bidang", end="")
            for j in range(6,maxbidang):
                print(" ", end="")
            print(" | partai", end="")
            for j in range(6,maxpartai):
                print(" ", end="")
            print(" |", end="")
            print()
            for j in range(maxNama+maxid+maxbidang+maxpartai+13):
                print("-", end="")
            print()
            
            for i in range(len(self.__listHumans)):
                print("| " + self.__listHumans[i].getId(), end="")
                for j in range(len(self.__listHumans[i].getId()),maxid):
                    print(" ", end="")
                print( " | " + self.__listHumans[i].getNama(), end="")
                for j in range(len(self.__listHumans[i].getNama()),maxNama):
                    print(" ", end="")
                print(" | " + self.__listHumans[i].getBidang(), end="")
                for j in range(len(self.__listHumans[i].getBidang()),maxbidang):
                    print(" ", end="")
                print(" | " + self.__listHumans[i].getPartai(), end="")
                for j in range(len(self.__listHumans[i].getPartai()),maxpartai):
                    print(" ", end="")
                print(" |", end="")
                print("\n", end="")
                for j in range(maxNama+maxid+maxbidang+maxpartai+13):
                    print("-", end="")
                print()
            
        
        print()
    

File: java/Crud.java
import java.util.Scanner;
import java.util.ArrayList;

public class Crud
{
	private ArrayList<Human> listHumans = new ArrayList<>();
	private Scanner sc = new Scanner(System.in);
	Crud(){
	}
	
	public void tambah(){
		String nama, id, bidang, partai;
		System.out.print("Masukkan id       : ");
		id = sc.next();
		System.out.print("Masukkan nama     : ");
		nama = sc.next();
		System.out.print("Masukkan bidang   : ");
		bidang = sc.next();
		System.out.print("Masukkan partai   : ");
		partai = sc.next();
		Human temp = new Human(id,nama,bidang,partai);
		listHumans.add(temp);
		System.out.println(nama + " berhasil ditambahkan!");
		System.out.println();
	}
	
	public void ubah(){
		String nama, id, bidang, partai;
		if(listHumans.size()==0){
			System.out.println("Data tidak ada");
		}else{
			System.out.println("Masukkan indeks nama");
			for(int i=0;i<listHumans.size();i++){
				System.out.println(Integer.toString(i+1)+
				". "+listHumans.get(i).getName());
			}
			System.out.println();
			int masukkan;
			try{
				masukkan = sc.nextInt();
				System.out.println();
			}catch(Exception e){
				masukkan=-1;
			}
			if(masukkan<=listHumans.size() && masukkan > 0){
				System.out.println("Masukkan indeks isi");
				System.out.println("1. id");
				System.out.println("2. nama");
				System.out.println("3. bidang");
				System.out.println("4. partai");
				System.out.println();
				
				int ubahB;
				try{
					ubahB = sc.nextInt();
					System.out.println("");
				}catch(Exception e){
					ubahB = -1;
				}
				switch(ubahB){
					case 1:
                        System.out.print("Masukkan id : ");
                        id = sc.next();
                        listHumans.get(masukkan-1).setId(id);
                        break;
                    case 2:
						System.out.print("Masukkan nama : ");
						nama = sc.next();
						listHumans.get(masukkan-1).setName(nama);
						break;
						
					case 3:
						System.out.print("Masukkan bidang : ");
						bidang = sc.next();
						listHumans.get(masukkan-1).setBidang(bidang);
						break;
					case 4:
						System.out.print("Masukkan partai : ");
						partai = sc.next();
						listHumans.get(masukkan-1).setPartai(partai);
						break;
					default:
						System.out.println("Indeks tidak valid");
				}
			}else{
				System.out.println("Indeks tidak valid");
			}
		}
		System.out.println();
	}
	
	public void hapus(){
		if(listHumans.size()==0){
			System.out.println("Data tidak ada");
		}else{
			System.out.println("Masukkan indeks nama");
			for(int i=0;i<listHumans.size();i++){
				System.out.println(Integer.toString(i+1)+
				". "+listHumans.get(i).getName());
			}
			System.out.println();
			int masukkan;
			try{
				masukkan = sc.nextInt();
				System.out.println();
			}catch(Exception e){
				masukkan = -1;
			}
			if(masukkan<=listHumans.size() && masukkan >0){
				System.out.println(listHumans.get(masukkan-1).getName() + " berhasil dihapus!");
				listHumans.remove(masukkan-1);
			}else{
				System.out.println("Indeks tidak valid");
			}
		}
		System.out.println();
	}
	
	public void tampil(){
		if(listHumans.size()==0){
			System.out.println("Data tidak ada");
		}else{
			int maxNama=4,maxid=2,maxbidang=6,maxpartai=6;
			for(int i=0;i<listHumans.size();i++){
                if(listHumans.get(i).getId().length()>maxid){
                    maxid = listHumans.get(i).getId().length();
				}
                if(listHumans.get(i).getName().length()>maxNama){
                    maxNama = listHumans.get(i).getName().length();
                }
				if(listHumans.get(i).getBidang().length()>maxbidang){
					maxbidang = listHumans.get(i).getBidang().length();
				}
				if(listHumans.get(i).getPartai().length()>maxpartai){
					maxpartai = listHumans.get(i).getPartai().length();
				}
			}
			for(int j=0;j<(maxNama+maxid+maxbidang+maxpartai+13);j++)System.out.print("-");
			System.out.println();
			
			System.out.print("| id");
			for(int j=2;j<maxid;j++)System.out.print(" ");
			System.out.print(" | Nama");
			for(int j=4;j<maxNama;j++)System.out.print(" ");
			System.out.print(" | bidang");
			for(int j=6;j<maxbidang;j++)System.out.print(" ");
			System.out.print(" | partai");
			for(int j=6;j<maxpartai;j++)System.out.print(" ");
			System.out.print(" |");
			System.out.println();
			for(int j=0;j<(maxid+maxNama+maxbidang+maxpartai+13);j++)System.out.print("-");
			System.out.println();
			
			for(int i=0;i<listHumans.size();i++){
                System.out.print("| "+listHumans.get(i).getId());
				for(int j=listHumans.get(i).getId().length();j<maxid;j++)System.out.print(" ");
				System.out.print(" | "+listHumans.get(i).getName());
				for(int j=listHumans.get(i).getName().length();j<maxNama;j++)System.out.print(" ");
				System.out.print(" | "+listHumans.get(i).getBidang());
				for(int j=listHumans.get(i).getBidang().length();j<maxbidang;j++)System.out.print(" ");
				System.out.print(" | "+listHumans.get(i).getPartai());
				for(int j=listHumans.get(i).getPartai().length();j<maxpartai;j++)System.out.print(" ");
				System.out.print(" |");
				System.out.println();
				for(int j=0;j<(maxNama+maxid+maxbidang+maxpartai+13);j++)System.out.print("-");
				System.out.println();
			}
		}
		System.out.println();
	}
	
}